compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -73,7 +73,20 @@ impl<'a> Parser<'a> {
             });
         }
 
-        let stmt = if self.token.is_keyword(kw::Let) {
+        let stmt = if self.token.is_keyword(kw::Super) && self.is_keyword_ahead(1, &[kw::Let]) {
+            self.collect_tokens(None, attrs, force_collect, |this, attrs| {
+                this.expect_keyword(exp!(Super))?;
+                this.psess.gated_spans.gate(sym::super_let, this.prev_token.span);
+                this.expect_keyword(exp!(Let))?;
+                let local = this.parse_local(attrs)?; // FIXME(mara): implement super let
+                let trailing = Trailing::from(capture_semi && this.token == token::Semi);
+                Ok((
+                    this.mk_stmt(lo.to(this.prev_token.span), StmtKind::Let(local)),
+                    trailing,
+                    UsePreAttrPos::No,
+                ))
+            })?
+        } else if self.token.is_keyword(kw::Let) {
```

Comment:
Since this commit adds the parsing, let's also please add a test for pretty printing and any changes needed for that.

Reply:
This PR purposely does the absolute minimum to add the feature gate. It adds parsing because CI enforces having a test that triggers the need for the new feature gate, so there needs to be a syntax that is only accepted with the gate.

However, this PR doesn't yet do anything with the parsed `super let`. It doesn't even store the fact that this `let` was `super` in the let statement. So there's nothing that can be pretty printed (yet).

That's all part of https://github.com/rust-lang/rust/pull/139112

I can try to move parts of it to this PR if you want, but I don't think that's very useful.

Reply:
OK, fair enough.